src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureBlobAction.java

```diff
@@ -75,77 +78,77 @@ public void onLoad() {
-	String blobName = queryPath.substring(1);
-
-	// Check the archive blob if it is non-null
-	if (zipArchiveBlob != null && zipArchiveBlob.getBlobName().equals(blobName)) {
-	    try {
-		response.sendRedirect2(zipArchiveBlob.getBlobURL() + "?"
-			+ WAStorageClient.generateSASURL(accountInfo, containerName, blobName));
-	    } catch (Exception e) {
-		response.sendError(500, "Error occurred while downloading artifact " + e.getMessage());
-	    }
-	    return;
-	}
-
-	for (AzureBlob blob : individualBlobs) {
-	    if (blob.getBlobName().equals(blobName)) {
-		try {
-		    response.sendRedirect2(blob.getBlobURL() + "?"
-			    + WAStorageClient.generateSASURL(accountInfo, containerName, blobName));
-		} catch (Exception e) {
-		    response.sendError(500, "Error occurred while downloading artifact " + e.getMessage());
-		}
-		return;
-	    }
-	}
-
-	response.sendError(404, "Azure artifact is not available");
+        AzureCredentials.StorageAccountCredential accountCredentials = AzureCredentials.getStorageCreds(storageCredentialId, storageAccountName);
+
+        if (accountCredentials == null) {
+            response.sendError(Constants.HTTP_INTERNAL_SERVER_ERROR, "Azure Storage account global configuration is missing");
+            return;
+        }
+
+        StorageAccountInfo accountInfo = AzureCredentials.convertToStorageAccountInfo(accountCredentials);
+
+        if (!allowAnonymousAccess && isAnonymousAccess(Jenkins.getAuthentication())) {
+            String url = request.getOriginalRequestURI();
+            response.sendRedirect("/login?from=" + url);
+            return;
+        }
+
+        String queryPath = request.getRestOfPath();
+
+        if (queryPath == null) {
+            return;
+        }
+
+        String blobName = queryPath.substring(1);
+
+        // Check the archive blob if it is non-null
+        if (zipArchiveBlob != null && zipArchiveBlob.getBlobName().equals(blobName)) {
+            try {
+                response.sendRedirect2(zipArchiveBlob.getBlobURL() + "?"
+                        + WAStorageClient.generateSASURL(accountInfo, containerName, blobName));
+            } catch (Exception e) {
+                response.sendError(Constants.HTTP_INTERNAL_SERVER_ERROR, "Error occurred while downloading artifact " + e.getMessage());
+            }
+            return;
+        }
+
+        for (AzureBlob blob : individualBlobs) {
+            if (blob.getBlobName().equals(blobName)) {
+                try {
+                    response.sendRedirect2(blob.getBlobURL() + "?"
+                            + WAStorageClient.generateSASURL(accountInfo, containerName, blobName));
+                } catch (Exception e) {
+                    response.sendError(Constants.HTTP_INTERNAL_SERVER_ERROR, "Error occurred while downloading artifact " + e.getMessage());
+                }
+                return;
+            }
+        }
+
+        response.sendError(Constants.HTTP_NOT_FOUND, "Azure artifact is not available");
```

Comment:
Replace magic numbers (HTTP codes) with constant definition, as a requirement of the check style.